In [9]:
import pandas as pd
import numpy as np
import scipy.stats as stats

### 1. A physician is evaluating a new diet for her patients with a family history of heart disease. To test the effectiveness of this diet, 16 patients are placed on the diet for 6 months. Their weights and triglyceride levels are measured before and after the study, and the physician wants to know if either set of measurements has changed. (Data set: dietstudy.csv)

In [8]:
diet = pd.read_csv('C:/Python/Class/Exercise/Exercise/7. Basic Stats - Case Study-1/dietstudy.csv')
diet.head()

patid  age  gender  tg0  tg1  tg2  tg3  tg4  wgt0  wgt1  wgt2  wgt3  wgt4
0      1   45    Male  180  148  106  113  100   198   196   193   188   192
1      2   56    Male  139   94  119   75   92   237   233   232   228   225
2      3   50    Male  152  185   86  149  118   233   231   229   228   226
3      4   46  Female  112  145  136  149   82   179   181   177   174   172
4      5   64    Male  156  104  157   79   97   219   217   215   213   214

Parameters taken into considerations for weights are initial weight(wgt0) and final weight which is weight after 6 months(wgt4).
Parameters taken into considerations for triglyceride levels are initial triglyceride level(tg0) 
and final triglyceride level which is weight after 6 months(tg4).

In [41]:
print("The triglyceride levels of patients were {}".format(diet.tg0.mean()))
print("The final triglyceride levels of patients are {}".format(diet.tg4.mean()))
print("The weights of patients were {}".format(diet.wgt0.mean()))
print("The final weights of patients are {}".format(diet.wgt4.mean()))

The triglyceride levels of patients were 138.4375
The final triglyceride levels of patients are 124.375
The weights of patients were 198.375
The final weights of patients are 190.3125


### Define the Hypothesis:
####  Stage 1 (For triglyceride levels):
  **Ho**: initial triglyceride level(tg0) == final triglyceride level(tg4) 
  **Ha**: initial triglyceride level(tg0) <> final triglyceride level(tg4)
        
####  Stage 2 (For weights):
  **Ho**: initial weight(wgt0) == final weight(wgt4) 
  **Ha**: initial weight(wgt0) <> final weight(wgt4) 

Since we have to 2 compare 2 continuous variable of 2 stage we will use 2 sample T- test.

In [13]:
# Stage 1 (For triglyceride levels):
tg = stats.ttest_rel(a=diet.tg0,b=diet.tg4)  

In [14]:
tg.pvalue

0.24874946576903698

In [15]:
tg.pvalue > 0.05

# Since p value is > 0.05, we fail to reject NULL Hypothesis

True

In [16]:
# Stage 2 (For weights):
wt = stats.ttest_rel(a=diet.wgt0,b=diet.wgt4)

In [17]:
wt.pvalue

1.137689414996614e-08

In [18]:
wt.pvalue > 0.05

# Since p value is < 0.05, we reject NULL Hypothesis

False

Conclusion : 
    
    Change in triglyceride levels, before and after diet is due to diet.
    In terms of Weights, the measurement before and after new diet is changed due to diet.

### 2. An analyst at a department store wants to evaluate a recent credit card promotion. To this end, 500 cardholders were randomly selected. Half received an ad promoting a reduced interest rate on purchases made over the next three months, and half received a standard seasonal ad. Is the promotion effective to increase sales? (Data set: creditpromo.csv)

In [19]:
credit = pd.read_csv('C:/Python/Class/Exercise/Exercise/7. Basic Stats - Case Study-1/creditpromo.csv')
credit.head()

id         insert      dollars
0   148       Standard  2232.771979
1   572  New Promotion  1403.807542
2   973       Standard  2327.092181
3  1096       Standard  1280.030541
4  1541  New Promotion  1513.563200

In [24]:
standard = credit[credit['insert'] == 'Standard']['dollars']
newPromotion = credit[credit['insert'] == 'New Promotion']['dollars']

In [33]:
print("The average spent by the normal standard seasonal ad is ${}".format(standard.mean()))

print("The average spent by the normal ad is ${}".format(newPromotion.mean()))

The average spent by the normal standard seasonal ad is $1566.3890309659348
The average spent by the normal ad is $1637.4999830647992


#### Step 1 : Define the null and alt hypothesis
**H0** : Sales Because of Promotion == Sales Beacuase of Seasonal Ad 
**H1** : Sales Because of Promotion > Sales Because of Seasonal Ad

insert - binomial categorical valiable(has 2 values)
dollars - continuous variable

Stats Method : t-test (Independent Sample T test). We are using T-test because we have 2 sample and as insert is binary categorical variable we will be considering this T-Test as independent T -Test.
Also test will have 2 assumption.
1) Standard and New Promotion have equal variance
2) Standard and New Promotion have different variance

If T value from both the test comes same we will consider both Standard and New Promotion ad have same variance else will consider the result are from different variance.

In [27]:
#equal variance
eq_var = stats.ttest_ind(a= standard, b= newPromotion, equal_var=True)
eq_var.statistic # fetch T value

-2.2604227264649963

In [30]:
#unequal variance
uneq_var = stats.ttest_ind(a= standard, b= newPromotion, equal_var=False)
uneq_var.statistic # fetch T value

-2.260422726464996

In [31]:
# We'll cosider equal variance since the t score is not having a huge difference
uneq_var.statistic - eq_var.statistic

4.440892098500626e-16

In [32]:
# Step 4: Since p_vale is less than 0.05,we reject the NULL hypothesis
eq_var.pvalue > 0.05

False

Conclusion: Since the significance value of the test is less than 0.05, we can conclude that promotional ad was effective to increase sales.

### 3. An experiment is conducted to study the hybrid seed production of bottle gourd under open field conditions. The main aim of the investigation is to compare natural pollination and hand pollination. The data are collected on 10 randomly selected plants from each of natural pollination and hand pollination. The data are collected on fruit weight (kg), seed yield/plant (g) and seedling length (cm). (Data set: pollination.csv)
### a. Is the overall population of Seed yield/plant (g) equals to 200?

In [35]:
pollination = pd.read_csv('C:/Python/Class/Exercise/Exercise/7. Basic Stats - Case Study-1/pollination.csv')
pollination.head()

Group  Fruit_Wt  Seed_Yield_Plant  Seedling_length
0  Natural      1.85            147.70            16.86
1  Natural      1.86            136.86            16.77
2  Natural      1.83            149.97            16.35
3  Natural      1.89            172.33            18.26
4  Natural      1.80            144.46            17.90

Define the hypothesis: 
    **Ho**: seed yield/plant == 200 
    **Ha**: seed yield/plant <> 200
 
Since, we are comparing Seed yield/plant with a value provided, so statistical method used is One Sample T-Test

In [37]:
# one sample t-test
test = stats.ttest_1samp(a=pollination.Seed_Yield_Plant,popmean=2000)

t_score = round(test.statistic,2)
p_value = test.pvalue

# CI : 0.95 or 95%
# alpha = 0.05
# if p value < alpha then rject NULL and 
# if it is >= alpha, we fail to reject NULL

print("The p value is {} and the T Score is {}".format(p_value,t_score))

The p value is 9.350752673732562e-34 and the T Score is -218.05


In [39]:
p_value > 0.05

# Since the significance value of the test is less than 0.05, we reject the null hypothesis.

False

Conclusion: Overall population of Seed yield/plant (g) is not equals to 200.

### b. Test whether the natural pollination and hand pollination under open field conditions are equally effective or are significantly different.

In [40]:
# Parameters taken into consideration are fruit weight (kg), seed yield/plant (g) and seedling length (cm) of 
# natural pollination and hand pollination.

natural = pollination[pollination['Group']=='Natural']
hand = pollination[pollination['Group']=='Hand']

### 4. An electronics firm is developing a new DVD player in response to customer requests. Using a prototype, the marketing team has collected focus data for different age groups viz. Under 25; 25-34; 35-44; 45-54; 55-64; 65 and above. Do you think that consumers of various ages rated the design differently? (Data set: dvdplayer.csv).

In [52]:
dvd = pd.read_csv('C:/Python/Class/Exercise/Exercise/7. Basic Stats - Case Study-1/dvdplayer.csv')
dvd.head()

agegroup   dvdscore
0  65 and over  38.454803
1        55-64  17.669677
2  65 and over  31.704307
3  65 and over  25.924460
4     Under 25  30.450007

In [58]:
grp1 = dvd[dvd['agegroup']== 'Under 25']['dvdscore']
grp2 = dvd[dvd['agegroup']== '25-34']['dvdscore']
grp3 = dvd[dvd['agegroup']== '35-44']['dvdscore']
grp4 = dvd[dvd['agegroup']== '45-54']['dvdscore']
grp5 = dvd[dvd['agegroup']== '55-64']['dvdscore']
grp6 = dvd[dvd['agegroup']== '65 and over']['dvdscore']

In [59]:
# Perform the ANOVA
aov = stats.f_oneway(grp1,grp2,grp3,grp4,grp5,grp6)

# Statistic :  F Value
print("The f-value is {} and the p value is {}".format(aov.statistic,aov.pvalue))

The f-value is 6.992526962676518 and the p value is 3.087324905679639e-05


In [61]:
aov.pvalue > 0.05

# Since p_value is less than 0.05 we reject the NULL Hypothesis

False

 Conclusion: consumers of various ages rated the design differently.

### 5. 5. A survey was conducted among 2800 customers on several demographic characteristics. Working status, sex, age, age-group, race, happiness, no. of child, marital status, educational qualifications, income group etc. had been captured for that purpose. (Data set: sample_survey.csv).
### a. Is there any relationship in between labour force status with marital status?

In [45]:
survey = pd.read_csv('C:/Python/Class/Exercise/Exercise/7. Basic Stats - Case Study-1/sample_survey.csv')
survey.head()

id            wrkstat        marital  childs   age  educ  paeduc  maeduc  \
0   1  Working full time       Divorced     2.0  60.0  12.0    12.0    12.0   
1   2  Working part-time  Never married     0.0  27.0  17.0    20.0     NaN   
2   3  Working full time        Married     2.0  36.0  12.0    12.0    12.0   
3   4  Working full time  Never married     0.0  21.0  13.0     NaN    12.0   
4   5  Working full time  Never married     0.0  35.0  16.0     NaN    12.0   

   speduc          degree  ...        agecat childcat news1 news2  news3  \
0     NaN     High school  ...      55 to 64      1-2    No    No     No   
1     NaN  Junior college  ...      25 to 34     None    No    No    Yes   
2    16.0     High school  ...      35 to 44      1-2    No    No     No   
3     NaN     High school  ...  Less than 25     None    No    No     No   
4     NaN        Bachelor  ...      35 to 44     None    No    No     No   

  news4 news5      car1      car2      car3  
0    No    No  American  Japanese  Japanese  
1    No    No  American    German  Japanese  
2   Yes   Yes  American  American       NaN  
3   Yes   Yes  American     Other       NaN  
4    No    No  American  American    Korean  

[5 rows x 46 columns]

In [63]:
cat_seg_xtab = pd.crosstab(survey.wrkstat, survey.marital, margins = True)
cat_seg_xtab 

marital                  Divorced  Married  Never married  Separated  Widowed  \
wrkstat                                                                         
Keeping house                  25      200             35         13       55   
Other                          12       16             14          4        8   
Retired                        53      168             17          6      150   
School                          7        9             60          2        1   
Temporarily not working         9       23             11          1        2   
Unemployed, laid off           10       13             32          0        3   
Working full time             295      778            392         58       44   
Working part-time              35      138            102          9       20   
All                           446     1345            663         93      283   

marital                   All  
wrkstat                        
Keeping house             328  
Other                      54  
Retired                   394  
School                     79  
Temporarily not working    46  
Unemployed, laid off       58  
Working full time        1567  
Working part-time         304  
All                      2830

In [64]:
x2test = stats.chi2_contingency(observed= cat_seg_xtab)
print("The chi square stat is {} and the p value is {}".format(x2test[0],x2test[1]))

The chi square stat is 729.2421426572284 and the p value is 1.820339965538765e-127


In [65]:
x2test[1] > 0.05

#Since p_value is less than 0.05 we reject the null hypothesis. 

False

Conclusion : There is a realtion between labour force status and marital status

### b. Do you think educational qualification is somehow controlling the marital status?

degree : Categorical
marital status : Categorical
    
as both variable are categorical variables and we have to find is there any relation betwen them. 
So we will perform chi square test

H0 : Observed == Expected (No relation between degree and marital status)
H1 : Observed != Expected (There is a realtionship between degree and marital status)

In [66]:
edu_cat_seg_xtab = pd.crosstab(survey.degree, survey.marital, margins = True)
edu_cat_seg_xtab 

marital         Divorced  Married  Never married  Separated  Widowed   All
degree                                                                    
Bachelor              58      251            129         12       28   478
Graduate              29      123             41          3        9   205
High school          241      686            367         58      148  1500
Junior college        45      108             46          3        6   208
LT High school        70      174             77         17       92   430
All                  443     1342            660         93      283  2821

In [67]:
x2test = stats.chi2_contingency(observed= edu_cat_seg_xtab)
print("The chi square stat is {} and the p value is {}".format(x2test[0],x2test[1]))

The chi square stat is 122.68449020508541 and the p value is 7.424404099753273e-15


In [68]:
x2test[1] > 0.05

#Since p_value is less than 0.05 we reject the null hypothesis. 

False

Conclusion: educational qualification is somehow controlling the marital status

### c. Is happiness is driven by earnings or marital status?

#### Test 1

In [79]:
earning_hap_xtab = pd.crosstab(survey.happy, survey.income, margins = True)
earning_hap_xtab 

income         $1000 TO 2999  $10000 - 14999  $15000 - 19999  $20000 - 24999  \
happy                                                                          
Not too happy              7              39              33              40   
Pretty happy              20             107             119             155   
Very happy                 5              44              26              50   
All                       32             190             178             245   

income         $25000 or more  $3000 TO 3999  $4000 TO 4999  $5000 TO 5999  \
happy                                                                        
Not too happy             113              9              9              6   
Pretty happy              888             11             13             18   
Very happy                571              4             10             11   
All                      1572             24             32             35   

income         $6000 TO 6999  $7000 TO 7999  $8000 TO 9999  LT $1000   All  
happy                                                                       
Not too happy             14             12              9        11   302  
Pretty happy              13             21             30        13  1408  
Very happy                 6             14             19        11   771  
All                       33             47             58        35  2481

In [84]:
x2test1 = stats.chi2_contingency(observed= earning_hap_xtab)
print("The chi square stat is {} and the p value is {}".format(x2test1[0],x2test1[1]))

The chi square stat is 178.95053061216427 and the p value is 7.234749067043371e-21


In [85]:
x2test[1] > 0.05

False

Since pvalue is less, we reject the NULL Hypothesis which means happiness is influenced by earning.

#### Test 2

In [82]:
marital_hap_xtab = pd.crosstab(survey.happy, survey.marital, margins = True)
marital_hap_xtab 

marital        Divorced  Married  Never married  Separated  Widowed   All
happy                                                                    
Not too happy        72       71            108         30       59   340
Pretty happy        278      684            426         49      137  1574
Very happy           93      582            120         13       83   891
All                 443     1337            654         92      279  2805

In [86]:
x2test2 = stats.chi2_contingency(observed= marital_hap_xtab)
print("The chi square stat is {} and the p value is {}".format(x2test2[0],x2test2[1]))

The chi square stat is 260.68943894182826 and the p value is 7.762777322980048e-47


In [87]:
x2test[1] > 0.05

False

Since pvalue is less, we reject the NULL Hypothesis which means happiness is influenced by marital status.

Conclusion : As seen above, both earning and marital status influence happiness.
But chi_square score for (happiness w.r.t to marital status) i.e. 260.68 is more than chi_square score for (happiness w.r.t to marital earnings) i.e 178.95, therefore we can conclude that happiness is more driven by marital status.